### ISSUE TITLE
Fails to parse JSON Schema from Google

### ISSUE THREAD

**Issue by Brian-McBride** [StefanTerdell/json-schema-to-zod#36]
# Unable to convert JSONSchemas from googleapis repository

## Description

I am trying to batch process the json schema from Google's source:

### Example source

https://github.com/googleapis/google-cloudevents/blob/main/jsonschema/google/events/cloud/apigateway/v1/GatewayEventData.json

## Problem output

This only generates the following:
```typescript
import { z } from "zod";

export default z.any();
```

### Example JSON (from link above)

```json
{
  "$id": "https://googleapis.github.io/google-cloudevents/jsonschema/google/events/cloud/apigateway/v1/GatewayEventData.json",
  "name": "GatewayEventData",
  "examples": [],
  "package": "google.events.cloud.apigateway.v1",
  "datatype": "google.events.cloud.apigateway.v1.GatewayEventData",
  "cloudeventTypes": [
    "google.cloud.apigateway.gateway.v1.created",
    "google.cloud.apigateway.gateway.v1.updated",
    "google.cloud.apigateway.gateway.v1.deleted"
  ],
  "product": "API Gateway",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "$ref": "#/definitions/GatewayEventData",
  "definitions": {
    "GatewayEventData": {
      "properties": {
        "payload": {
          "$ref": "#/definitions/Gateway",
          "additionalProperties": true,
          "description": "Optional. The Gateway event payload. Unset for deletion events."
        }
      },
      "additionalProperties": true,
      "type": "object",
      "title": "Gateway Event Data",
      "description": "The data within all Gateway events."
    },
    "Gateway": {
      "properties": {
        "name": {
          "type": "string",
          "description": "Output only. Resource name of the Gateway. Format: projects/{project}/locations/{location}/gateways/{gateway}"
        },
        "createTime": {
          "type": "string",
          "description": "Output only. Created time.",
          "format": "date-time"
        },
        "updateTime": {
          "type": "string",
          "description": "Output only. Updated time.",
          "format": "date-time"
        },
        "labels": {
          "additionalProperties": {
            "type": "string"
          },
          "type": "object",
          "description": "Optional. Resource labels to represent user-provided metadata. Refer to cloud documentation on labels for more details. https://cloud.google.com/compute/docs/labeling-resources"
        },
        "displayName": {
          "type": "string",
          "description": "Optional. Display name."
        },
        "apiConfig": {
          "type": "string",
          "description": "Required. Resource name of the API Config for this Gateway. Format: projects/{project}/locations/global/apis/{api}/configs/{apiConfig}"
        },
        "state": {
          "enum": [
            "STATE_UNSPECIFIED",
            0,
            "CREATING",
            1,
            "ACTIVE",
            2,
            "FAILED",
            3,
            "DELETING",
            4,
            "UPDATING",
            5
          ],
          "oneOf": [
            {
              "type": "string"
            },
            {
              "type": "integer"
            }
          ],
          "title": "State",
          "description": "All the possible Gateway states."
        },
        "defaultHostname": {
          "type": "string",
          "description": "Output only. The default API Gateway host name of the form `{gateway_id}-{hash}.{region_code}.gateway.dev`."
        }
      },
      "additionalProperties": true,
      "type": "object",
      "title": "Gateway",
      "description": "A Gateway is an API-aware HTTP proxy. It performs API-Method and/or API-Consumer specific actions based on an API Config such as authentication, policy enforcement, and backend selection."
    }
  }
}
```

**Comment by StefanTerdell** [comment]
Once again seems like an issue with dereferencing using @apidevtools/json-schema-ref-parser, so seems related to #29 to me - if you replace the root $ref with its target schema it seems to work alright. SMH

**Comment by Brian-McBride** [comment]
I see that. I wrote a script to reformat the JSON. 
However, on one file I got : `Maximum call stack size exceeded`

I'm including the JSON below in the org format. Any idea if this is part of the ref-parser? I'm sure it is the circular reference there. 

```json
{
  "$id": "https://googleapis.github.io/google-cloudevents/jsonschema/google/events/cloud/beyondcorp/appconnectors/v1/AppConnectorEventData.json",
  "name": "AppConnectorEventData",
  "examples": [],
  "package": "google.events.cloud.beyondcorp.appconnectors.v1",
  "datatype": "google.events.cloud.beyondcorp.appconnectors.v1.AppConnectorEventData",
  "cloudeventTypes": [
    "google.cloud.beyondcorp.appconnectors.appConnector.v1.created",
    "google.cloud.beyondcorp.appconnectors.appConnector.v1.updated",
    "google.cloud.beyondcorp.appconnectors.appConnector.v1.deleted"
  ],
  "product": "BeyondCorp",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "$ref": "#/definitions/AppConnectorEventData",
  "definitions": {
    "AppConnectorEventData": {
      "properties": {
        "payload": {
          "$ref": "#/definitions/AppConnector",
          "additionalProperties": true,
          "description": "Optional. The AppConnector event payload. Unset for deletion events."
        }
      },
      "additionalProperties": true,
      "type": "object",
      "title": "App Connector Event Data",
      "description": "The data within all AppConnector events."
    },
    "AppConnector": {
      "properties": {
        "name": {
          "type": "string",
          "description": "Required. Unique resource name of the AppConnector. The name is ignored when creating a AppConnector."
        },
        "createTime": {
          "type": "string",
          "description": "Output only. Timestamp when the resource was created.",
          "format": "date-time"
        },
        "updateTime": {
          "type": "string",
          "description": "Output only. Timestamp when the resource was last modified.",
          "format": "date-time"
        },
        "labels": {
          "additionalProperties": {
            "type": "string"
          },
          "type": "object",
          "description": "Optional. Resource labels to represent user provided metadata."
        },
        "displayName": {
          "type": "string",
          "description": "Optional. An arbitrary user-provided name for the AppConnector. Cannot exceed 64 characters."
        },
        "uid": {
          "type": "string",
          "description": "Output only. A unique identifier for the instance generated by the system."
        },
        "state": {
          "enum": [
            "STATE_UNSPECIFIED",
            0,
            "CREATING",
            1,
            "CREATED",
            2,
            "UPDATING",
            3,
            "DELETING",
            4,
            "DOWN",
            5
          ],
          "oneOf": [
            {
              "type": "string"
            },
            {
              "type": "integer"
            }
          ],
          "title": "State",
          "description": "Represents the different states of a AppConnector."
        },
        "principalInfo": {
          "$ref": "#/definitions/PrincipalInfo",
          "additionalProperties": true,
          "description": "Required. Principal information about the Identity of the AppConnector."
        },
        "resourceInfo": {
          "$ref": "#/definitions/ResourceInfo",
          "additionalProperties": true,
          "description": "Optional. Resource info of the connector."
        }
      },
      "additionalProperties": true,
      "type": "object",
      "title": "App Connector",
      "description": "A BeyondCorp connector resource that represents an application facing component deployed proximal to and with direct access to the application instances. It is used to establish connectivity between the remote enterprise environment and GCP. It initiates connections to the applications and can proxy the data from users over the connection."
    },
    "PrincipalInfo": {
      "properties": {
        "serviceAccount": {
          "$ref": "#/definitions/ServiceAccount",
          "additionalProperties": true,
          "description": "A GCP service account."
        }
      },
      "additionalProperties": true,
      "type": "object",
      "title": "Principal Info",
      "description": "PrincipalInfo represents an Identity oneof."
    },
    "ServiceAccount": {
      "properties": {
        "email": {
          "type": "string",
          "description": "Email address of the service account."
        }
      },
      "additionalProperties": true,
      "type": "object",
      "title": "Service Account",
      "description": "ServiceAccount represents a GCP service account."
    },
    "ResourceInfo": {
      "properties": {
        "id": {
          "type": "string",
          "description": "Required. Unique Id for the resource."
        },
        "status": {
          "enum": [
            "HEALTH_STATUS_UNSPECIFIED",
            0,
            "HEALTHY",
            1,
            "UNHEALTHY",
            2,
            "UNRESPONSIVE",
            3,
            "DEGRADED",
            4
          ],
          "oneOf": [
            {
              "type": "string"
            },
            {
              "type": "integer"
            }
          ],
          "title": "Health Status",
          "description": "HealthStatus represents the health status."
        },
        "time": {
          "type": "string",
          "description": "The timestamp to collect the info. It is suggested to be set by the topmost level resource only.",
          "format": "date-time"
        },
        "sub": {
          "items": {
            "$ref": "#/definitions/ResourceInfo"
          },
          "type": "array",
          "description": "List of Info for the sub level resources."
        }
      },
      "additionalProperties": true,
      "type": "object",
      "title": "Resource Info",
      "description": "ResourceInfo represents the information/status of an app connector resource. Such as: - remote_agent   - container     - runtime     - appgateway       - appconnector         - appconnection           - tunnel       - logagent"
    }
  }
}
```

**Comment by StefanTerdell** [comment]
Fixed in 1.0.0
